sklearn_numba_dpex/tests/test_kmeans.py

```diff
@@ -89,45 +94,64 @@ def test_kmeans_same_results(dtype):
 
 
 @pytest.mark.parametrize("dtype", [np.float32, np.float64])
-def test_kmeans_fit_empty_clusters(dtype):
+def test_kmeans_relocated_clusters(dtype):
+    """Copied and adapted from sklearn's unit tests"""
     _fail_if_no_dtype_support(pytest.xfail, dtype)
 
-    random_seed = 42
-    X, _ = make_blobs(random_state=random_seed)
-    X = X.astype(dtype)
+    # check that empty clusters are relocated as expected
+    X = np.array([[0, 0], [0.5, 0], [0.5, 1], [1, 1]], dtype=dtype)
+
+    # second center too far from others points will be empty at first iter
+    init_centers = np.array([[0.5, 0.5], [3, 3]], dtype=dtype)
+
+    kmeans = KMeans(n_clusters=2, n_init=1, init=init_centers)
+    with config_context(engine_provider="sklearn_numba_dpex"):
+        kmeans.fit(X)
+
+    expected_n_iter = 3
+    expected_inertia = 0.25
+    assert_allclose(kmeans.inertia_, expected_inertia)
+    assert kmeans.n_iter_ == expected_n_iter
+
+    # There are two acceptable ways of relocating clusters in this example, the output
+    # depends on how the argpartition strategy break ties. It might not be deterministic
+    # (might depend on thread concurrency) so we accept both outputs.
```

Comment:
What if we change this test to simultaneously sort both arrays prior to assertions?

I think this can be done by `argsort`-ing on `expected_centers` to get a permutation that we can apply on both `expected_labels` and `expected_centers`. What do you think of this?

Reply:
I don't think the fact that the results differs only by a permutation is not particularly interesting to use, it only holds for this particular example and would not be true in general.